spec/requests/providers/select_offices_spec.rb

```diff
@@ -1,8 +1,12 @@
 require 'rails_helper'
 
 RSpec.describe 'provider selects office', type: :request do
-  let(:provider) { create :provider, firm: firm }
   let(:firm) { create :firm }
+  let!(:office_1) { create :office, code: 'office1', firm: firm }
+  let!(:office_2) { create :office, code: 'office2', firm: firm }
+  let!(:office_3) { create :office, code: 'office3', firm: firm }
```

Comment:
I think you don't need to specify the code in these `let` statements as you are not using the code string in any matching (good!) but rather using matches on `office_<n>.code`. So:
```ruby
  let!(:office_1) { create :office, firm: firm }
  let!(:office_2) { create :office, firm: firm }
  let!(:office_3) { create :office, firm: firm }
```

Reply:
👍 
This was to make sure these code were unique.
But I'm removing them and replacing with with `Faker::Number.unique.number(10)` in the factory